app/Http/Requests/Three/PostRequest.php

```diff
@@ -27,11 +27,14 @@ public function rules()
             'campaign_id' => 'required',
             'campaign_run_id' => 'required_with:campaign_id|integer',
             'northstar_id' => 'nullable|objectid',
+            'type' => 'required|string',
+            'action' => 'required|string',
```

Comment:
Do we want to also validate format a bit here with [`alpha_dash`](https://laravel.com/docs/5.5/validation#rule-alpha-dash)?

Reply:
Sure! I did not know about that rule. Very nice.